sklearn/datasets/_samples_generator.py

```diff
@@ -1781,3 +1781,112 @@ def make_checkerboard(shape, n_clusters, *, noise=0.0, minval=10,
                       for label in range(n_col_clusters)])
 
     return result, rows, cols
+
+def make_er_graph(n, m, directed=False):
+    '''Generates an Erdos-Reyni (ER) graph.
+
+    Erdos Renyi (n, m) graph is a simple graph with n vertices and exactly m
+    number of total edges.
+
+    Parameters
+    ----------
+    n: int
+        Number of vertices
+    m: int
+        Number of edges, a value between 1 and :math:`n^2`.
+    directed: boolean, optional (default=False)
+        If False, output adjacency matrix will be symmetric. Otherwise, output adjacency
+        matrix will be asymmetric.
+
+    Returns
+    -------
+    A: ndarray, shape (n, n)
+        Sampled adjacency matrix
+
+    Notes
+    -----
+    Code annotated from microsoft/graspologic, simulations.er_nm
+    '''
+    A = np.zeros((n, n))
+    if directed:
+        idx = np.where(~np.eye(n, dtype=bool))
+    else:
+        idx = np.triu_indices(n, k=1)
+    # get idx in 1d coordinates by ravelling
+    triu = np.ravel_multi_index(idx, A.shape)
+    # choose M of them
+    triu = np.random.choice(triu, size=m, replace=False)
+    # unravel back
+    triu = np.unravel_index(triu, A.shape)
+    A[triu] = 1
+
+    if not directed:
+        A = np.triu(A)
+        A = A + A.T - np.diag(np.diag(A))
+    return A
+
+def make_sbm_graph(n, p, directed=False):
+    '''Generates Stochastic Block Model (SBM) graph
+
+    SBM produces a graph with specified communities, in which each community can
+    have different sizes and edge probabilities.
+
+    Parameters
+    ----------
+    n: list of int, shape (n_communities)
+        Number of vertices in each community. Communities are assigned n[0], n[1], ...
+    p: array-like, shape (n_communities, n_communities)
+        Probability of an edge between each of the communities, where ``p[i, j]`` indicates
+        the probability of a connection between edges in communities ``[i, j]``.
+        ``0 < p[i, j] < 1`` for all ``i, j``.
+    directed: boolean, optional (default=False)
+        If False, output adjacency matrix will be symmetric. Otherwise, output adjacency
+        matrix will be asymmetric.
+
+    Returns
+    -------
+    A: ndarray, shape (sum(n), sum(n))
+        Sampled adjacency matrix
+     Notes
+    -----
+    Code annotated from microsoft/graspologic, simulations.sbm
+    '''
```

Comment:
What do you mean by the notes here?

Small nitpick:
```suggestion

    Notes
    -----
    Code annotated from microsoft/graspologic, simulations.sbm
    '''
```

Reply:
the code for the [Erdos-Reyni](https://github.com/microsoft/graspologic/blob/c266aa3ff5a40d95bf4b4f88dc842b4b6551f7d4/graspologic/simulations/simulations.py#L171) and [Stochastic Block Model](https://github.com/microsoft/graspologic/blob/c266aa3ff5a40d95bf4b4f88dc842b4b6551f7d4/graspologic/simulations/simulations.py#L306) simulations are annotated from code in microsoft/graspologic since it is not a dependency of sk-learn. I simplified the code to fit the use case of the tests.

Reply:
> the code for the [Erdos-Reyni](https://github.com/microsoft/graspologic/blob/c266aa3ff5a40d95bf4b4f88dc842b4b6551f7d4/graspologic/simulations/simulations.py#L171) and [Stochastic Block Model](https://github.com/microsoft/graspologic/blob/c266aa3ff5a40d95bf4b4f88dc842b4b6551f7d4/graspologic/simulations/simulations.py#L306) simulations are annotated from code in microsoft/graspologic since it is not a dependency of sk-learn. I simplified the code to fit the use case of the tests.

The codes that generate graphs for benchmarking are very useful well beyond this PR. May be they can be placed in a better high-level place and be accessible for codes, e.g.,  in benchmarks? May be even a separate PR? 

If not yet there, please consider adding an option to generate the adjacency matrix in a sparse format, like in  https://github.com/ffrankies/GraphChallenge/blob/master/StochasticBlockPartition/data/graph_generator.py